### PR TITLE
Issue 8620 - deprecate `ConversionService.SHARED`

### LINKED ISSUE #8620: Deprecate `ConversionService.SHARED` in Micronaut 3.9, for removal in Micronaut 4
### Issue description

Deprecate in Micronaut 3.9, for removal in Micronaut 4

### PR DESCRIPTION
**Pull request by wetted**
closes #8620

### CODE REVIEW

**Comment by dstepanov** [comment]
Please annotate it with `@Deprecated`

**Review by alvarosanchez — CHANGES_REQUESTED** [review]
Also, this needs to target 3.9.x

**Comment by sdelamo** [comment]
> Also, this needs to target 3.9.x

Correct, the PR base should be `3.9.x` @wetted.

File: core/src/main/java/io/micronaut/core/convert/ConversionService.java [hunk 1]
```diff
@@ -31,6 +31,8 @@ public interface ConversionService {
 
     /**
      * The default shared conversion service.
+     *
+     * @Deprecated This will be removed in the next version. Use an injected {@link ConversionService} instead.
      */
     ConversionService SHARED = new DefaultMutableConversionService();
```
Comment (sdelamo) [inline core/src/main/java/io/micronaut/core/convert/ConversionService.java:38]: Annotate with `@Deprecated`.

**Review by sdelamo — CHANGES_REQUESTED** [review]
change the PR base to 3.9.x

**Pull request closed without merging**